Add BME680 BSEC docs

trvrnrth opened the pull request:
## Description:


**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#1313

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

trvrnrth committed:
Add BME680 BSEC docs

OttoWinter commented on components/sensor/bme680_bsec.rst:
```diff
@@ -0,0 +1,192 @@
+.. note::
+
+    The Bosch BSEC library is only available for use after accepting its software license agreement. By enabling this component,
+    you are explicitly agreeing to the terms of the `BSEC license agreement`_.
```
So if someone shares a build (the "`.bin`" file) with this component enabled, they would technically violate the ESPHome GPL license. Let's add a line saying that any firmware binary can't be shared when this component is enabled.

trvrnrth replied:
That sounds reasonable to me if that's what is required to meet the GPL requirements. My understanding of the licensing specifics for this scenario aren't really sufficient to give any input so on that basis I'm perfectly happy to add that as an extra restriction.

trvrnrth replied:
Added words to that effect in b1daada.

trvrnrth committed:
Disallow distribution of bins that include BSEC

In order to be safe in ahderence with the ESPHome GPL license add
a note that distribution of firmware binaries that include the
BSEC software is not allowed.

trvrnrth committed:
Update docs to reflect simplified state save

trvrnrth committed:
Update license link and simplify link to community

trvrnrth committed:
Typos

trvrnrth committed:
Update docs for latest features

Also include a full annotated advanced configuration example.

trvrnrth committed:
Drop support for multiple sensors

OttoWinter reviewed (approved):
Thanks! LGTM :+1:
